src/cockpit/transports.py

```diff
@@ -412,14 +412,18 @@ def kill(self) -> None:
         self.send_signal(signal.SIGKILL)
 
     def _close(self) -> None:
+        if self._pty_fd is not None:
+            os.close(self._pty_fd)
+            self._pty_fd = None
+
         if self._process is not None:
+            if self._process.stdin is not None:
```

Comment:
I think I'd have written out explicitly `if self._process is not None and self._process.stdin is not None` in a separate `if:` statement so that it was visually better matched up with the PTY case, but this is some serious bikeshedding at this point...